test/RouteMiddlewareTest.php

```diff
@@ -621,100 +621,29 @@ public function testInjectsRouteResultAsAttribute()
         $this->assertTrue($triggered);
     }
 
-    public function testMiddlewareTriggersObserversWithSuccessfulRouteResult()
+    /**
+     * @error-handling
+     */
+    public function testRoutingMiddlewareShouldReturn405ResponseDirectlyWhenRaiseThrowablesFlagEnabled()
     {
-        $matches    = ['id' => 'IDENTIFIER'];
-        $triggered  = false;
-        $middleware = function ($request, $response, $next) {
-            return $response;
-        };
-        $next = function ($request, $response, $err = null) {
-            $this->fail('Should not hit next');
-        };
-
         $request  = new ServerRequest();
         $response = new Response();
-        $result   = RouteResult::fromRouteMatch('resource', $middleware, $matches);
+        $result   = RouteResult::fromRouteFailure(['GET', 'POST']);
```

Comment:
I *think* I've updated all of those on the develop branch. There's a LOT of them, and, as noted, I'd rather not duplicate effort here, in particular to prevent merge issues later.